File: week2/time_print.c
#include <stdio.h>
#include <time.h>

int main() {
    clock_t start, end;
    FILE *file;

    file = fopen("output_c.txt", "w");
    if (file == NULL) {
        printf("Error opening file!\n");
        return 1;
    }

    start = clock(); // Start timing

    for (int i = 0; i < 1000000; i++) {
        fprintf(file, "Line %d\n", i);
    }

    end = clock(); // End timing
    fclose(file);

    double time_taken = ((double)(end - start)) / CLOCKS_PER_SEC;
    printf("C File Write Time: %.6f seconds\n", time_taken);

    return 0;
}

File: week2/time_print.py
import time

def main():
    start_time = time.time()  # Record start time

    with open("output_python.txt", "w") as file:
        for i in range(1000000):
            file.write(f"Line {i}\n")

    end_time = time.time()  # Record end time
    print(f"Python File Write Time: {end_time - start_time:.6f} seconds")

if __name__ == "__main__":
    main()
